### PR TITLE
mcuboot-imgtool: init at 1.10.0

### PR DESCRIPTION
**Pull request by otavio**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/imgtool/default.nix [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    click
+    cryptography
+    intelhex
+  ];
+
+  pythonImportsCheck = [ "imgtool" ];
+
+  meta = with lib; {
+    description = "MCUboot's image signing and key management";
+    homepage = "https://pypi.org/project/imgtool";
```
Comment (mweinelt) [inline pkgs/development/python-modules/imgtool/default.nix:43]:
```suggestion
    homepage = "https://github.com/mcu-tools/mcuboot/tree/main/scripts";
```

File: pkgs/development/python-modules/imgtool/default.nix [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  ];
+
+  propagatedBuildInputs = [
+    cbor2
+    click
+    cryptography
+    intelhex
+  ];
+
+  pythonImportsCheck = [ "imgtool" ];
```
Comment (mweinelt) [inline pkgs/development/python-modules/imgtool/default.nix:39]:
```suggestion
  pythonImportsCheck = [ "imgtool" ];
  
  doCheck = false;
```

No tests? Then disable the check phase.

File: pkgs/development/python-modules/imgtool/default.nix [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-A7NOdZNKw9lufEK2vK8Rzq9PRT98bybBfXJr0YMQS0A=";
+  };
+
+  passthru.updateScript = nix-update-script { };
+
+  nativeBuildInputs = [
+    setuptools
+    wheel
```
Comment (mweinelt) [inline pkgs/development/python-modules/imgtool/default.nix:29]:
```suggestion
```

Providing wheel here is not necessary.

File: pkgs/development/python-modules/imgtool/default.nix [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  pname = "imgtool";
+  version = "1.10.0";
+  pyproject = true;
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-A7NOdZNKw9lufEK2vK8Rzq9PRT98bybBfXJr0YMQS0A=";
+  };
+
+  passthru.updateScript = nix-update-script { };
```
Comment (mweinelt) [inline pkgs/development/python-modules/imgtool/default.nix:25]:
```suggestion
```

All python packages come with an update script by default.

File: pkgs/development/python-modules/imgtool/default.nix [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+, buildPythonPackage
+, nix-update-script
+
+, setuptools
+, wheel
+
+, cbor2
+, click
+, cryptography
+, intelhex
```
Comment (mweinelt) [inline pkgs/development/python-modules/imgtool/default.nix:12]:
```suggestion

# build-system
, setuptools

# dependencies
, cbor2
, click
, cryptography
, intelhex
```

**Review by mweinelt — DISMISSED** [review]

**Comment by mweinelt** [comment]
Is this really a library package? Looks like an application to me, in which case it should live outside the python package set and use `buildPythonApplication`.

File: pkgs/development/python-modules/imgtool/default.nix (hunk 4 above)
Comment (otavio) [reply pkgs/development/python-modules/imgtool/default.nix]: I moved it to an app package so I kept it.

**Comment by otavio** [comment]
ping?

**Pull request merged**